### ISSUE TITLE
[Feature Request] Add support for token-based light sources

### ISSUE THREAD

**Issue by FoxLee** [caewok/fvtt-light-mask#4]
I don't know if it's feasible because I'm not well-versed with Foundry's API—but, it would be fantastic if this module could be used to shape light sources generated by tokens as well as independent light sources. (I play D&D4e, where a radius is just a square in disguise.)

**Comment by caewok** [comment]
Fixed in version 0.3.0. Not sure if I can get the default token settings to do this as well, but I will look into it.

**Comment by FoxLee** [comment]
Wow, thanks for acting on this! While this seems to have the basics covered, there are a few quirks that could use attention if you have time.

The big issue is that the rotation setting seems to have no effect on token lights. No matter what I set it to, a regular polygon with four sides is fixed with its corners at the cardinal directions, so I can't make it look like a square rather than a diamond. This seems to be true for all shapes, and rotating the token doesn't help either.

I had more luck making a square using walls. However, when I set the walls to move relative to the token, it offsets the mask southeast by about half a square. So in the example below, if I apply the walls on the left with the token in that position, I get the mask shown on the right.

![image](https://user-images.githubusercontent.com/10067730/170414786-8f53b22c-b65f-43ee-a5e7-fcd4fea5ec87.png)

I _can_ approximate the right mask by moving my walls in the opposite direction before caching, but the result is a bit janky, and frustrating to do every time.

This is under the D&D 4e game system, with all other modules except libwrapper disabled.
